src/service/search/mod.rs

```diff
@@ -753,27 +762,17 @@ pub async fn search_partition(
         };
     }
 
-    // Generate partitions by DESC order
-    let mut partitions = Vec::with_capacity(part_num);
-    let mut end = req.end_time;
-    let mut last_partition_step = end % min_step;
-    let duration = req.end_time - req.start_time;
-    while end > req.start_time {
-        let mut start = max(end - step, req.start_time);
-        if last_partition_step > 0 && duration > min_step && part_num > 1 {
-            partitions.push([end - last_partition_step, end]);
-            start -= last_partition_step;
-            end -= last_partition_step;
-        } else {
-            start = max(start - last_partition_step, req.start_time);
-        }
-        partitions.push([start, end]);
-        end = start;
-        last_partition_step = 0;
-    }
-    if partitions.is_empty() {
-        partitions.push([req.start_time, req.end_time]);
-    }
+    let is_histogram = query.from == -1;
```

Comment:
is_histogram, you can't use `query.from` to decide, i think we have something related to it.

Reply:
Got it!